bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/PersistentInbox.java

```diff
@@ -173,43 +176,46 @@ public Thing approve(ThingUID thingUID, String label) {
     }
 
     @Override
-    public synchronized boolean add(DiscoveryResult result) throws IllegalStateException {
-        if (result != null) {
-            ThingUID thingUID = result.getThingUID();
-            Thing thing = this.thingRegistry.get(thingUID);
-
-            if (thing == null) {
-                DiscoveryResult inboxResult = get(thingUID);
-
-                if (inboxResult == null) {
-                    discoveryResultStorage.put(result.getThingUID().toString(), result);
-                    notifyListeners(result, EventType.added);
-                    logger.info("Added new thing '{}' to inbox.", thingUID);
+    public synchronized boolean add(final @Nullable DiscoveryResult rslt) throws IllegalStateException {
+        if (rslt == null) {
```

Comment:
Can we use `discoveryResult` instead of `rslt`?

Reply:
If the Eclipse Nullness checker would be clever (as others) it would not be necessary to introduce another local variable at all.
After returning if the argument is null, the checker should know that all occurrences in that method (at least as the argument is also final) are known to be non-null.
But the checker will complain about...

Sure, I don't care how to name the argument. The method signature is defined by the interface, also the JavaDoc, so the name here that is used for the nullness check at all should not matter.